packages/api/tests/endpoints/collections/delete-collection.js

```diff
@@ -126,3 +141,80 @@ test('Deleting a collection removes it', async (t) => {
     });
   });
 });
+
+test('Attempting to delete a collection with an associated rule returns a 400 response', async (t) => {
+  const collection = fakeCollectionFactory();
+  await collectionModel.create(collection);
+
+  const rule = fakeRuleFactoryV2({
+    collection: {
+      name: collection.name,
+      version: collection.version
+    },
+    rule: {
+      type: 'onetime'
+    }
+  });
+
+  // The workflow message template must exist in S3 before the rule can be created
+  await s3().putObject({
+    Bucket: process.env.bucket,
+    Key: `${process.env.stackName}/workflows/${rule.workflow}.json`,
+    Body: JSON.stringify({})
+  }).promise();
+
+  await ruleModel.create(rule);
+
+  const deleteRequest = {
+    httpMethod: 'DELETE',
+    pathParameters: {
+      collectionName: collection.name,
+      version: collection.version
+    },
+    headers: authHeaders
+  };
+
+  return testEndpoint(collectionsEndpoint, deleteRequest, (response) => {
+    t.is(response.statusCode, 400);
```

Comment:
How much do we care about the status code here? I briefly looked into it when setting this up on the fake API and I thought a 409 code might be appropriate, since it indicates that the request isn't bad but that the server cannot complete the action with its current configuration. I don't know if it matters, just flagging it.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409